src/main/java/com/amazon/opendistro/elasticsearch/performanceanalyzer/rca/framework/core/RcaConf.java

```diff
@@ -108,10 +110,41 @@ public int getPerVertexBufferLength() {
   }
 
   public HighHeapUsageOldGenRcaConfig getHighHeapUsageOldGenRcaConfig() {
-    return conf.getHighHeapUsageOldGenRcaConfig();
+    return new HighHeapUsageOldGenRcaConfig(this);
+  }
+
+  public HighHeapUsageYoungGenRcaConfig getHighHeapUsageYoungGenRcaConfig() {
+    return new HighHeapUsageYoungGenRcaConfig(this);
+  }
+
+  public HotNodeClusterRcaConfig getHotNodeClusterRcaConfig() {
+    return new HotNodeClusterRcaConfig(this);
   }
 
   public List<String> getMutedRcaList() {
     return conf.getMutedRcaList();
   }
+
+  @SuppressWarnings("unchecked")
+  public <T> T readRcaConfig(String rcaName, String key, Class<? extends T> clazz) {
+    T setting = null;
+    try {
+      Map<String, Object> rcaObj = null;
+      if (conf.getRcaConfigSettings() != null
+          && conf.getRcaConfigSettings().containsKey(rcaName)
+          && conf.getRcaConfigSettings().get(rcaName) != null) {
+        rcaObj = (Map<String, Object>)conf.getRcaConfigSettings().get(rcaName);
+      }
+
+      if (rcaObj != null
+          && rcaObj.containsKey(key)
+          && rcaObj.get(key) != null) {
+        setting = clazz.cast(rcaObj.get(key));
+      }
+    }
+    catch (ClassCastException ne) {
+      LOG.error("rca.conf contains value in invalid format, trace : {}", ne.getMessage());
```

Comment:
It might be difficult to know which of the casts threw the exception, line `136` or line `142`, right ?